src/reyna/Blackout/Time.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace Reyna
+{
+    using System;
+
+    public class Time
+    {
+        public Time(int hour, int minute)
+        {
+            if (hour >= 24 || minute >= 60 || hour < 0 || minute < 0)
+            {
+                throw new ArgumentException("Invalid time");
+            }
+
+            this.MinuteOfDay = (hour * 60) + minute;
+        }
+
+        public Time(int minuteOfDay)
+        {
+            if (minuteOfDay < 0 || minuteOfDay >= 1440)
+            {
+                throw new ArgumentException("Invalid minute of day");
+            }
+
+            this.MinuteOfDay = minuteOfDay;
+        }
+
+        public Time()
+        {
+            this.MinuteOfDay = (DateTime.UtcNow.Hour * 60) + DateTime.UtcNow.Minute;
+        }
+
+        public int MinuteOfDay { get; private set; }
+
+        public bool BeforeOrEqualTo(Time other)
+        {
+            return this.MinuteOfDay <= other.MinuteOfDay;
+        }
+
+        public bool AfterOrEqualTo(Time other)
+        {
+            return this.MinuteOfDay >= other.MinuteOfDay;
+        }
```

Comment:
Looks like implementation of IComparable<T> fits the purpose of `BeforeOrEqualTo` and `AfterOrEqualTo`
@ivanbokii